src/main/java/org/matsim/episim/model/DefaultInfectionModel.java

```diff
@@ -61,6 +69,27 @@ private void infectionDynamicsGeneralized(EpisimPerson personLeavingContainer, E
             int idx = rnd.nextInt(container.getPersons().size());
             EpisimPerson otherPerson = container.getPersons().get(idx);
 
+            String leavingPersonsActivity = personLeavingContainer.getTrajectory().get(personLeavingContainer.getCurrentPositionInTrajectory());
+            String otherPersonsActivity = otherPerson.getTrajectory().get(otherPerson.getCurrentPositionInTrajectory());
+
+            String infectionType = leavingPersonsActivity + "_" + otherPersonsActivity;
+
+            //forbid certain cross-activity infections, keep track of contacts
+            //we can not track contact persons in vehicles
+            if (infectionSituation.equals(InfectionSituation.Facility)){
+                //home can only interact with home or leisure
+                if (infectionType.contains("home") && ! infectionType.contains("leis")  && ! ( leavingPersonsActivity.contains("home") && otherPersonsActivity.contains("home")  )){
+                    continue;
+                } else if (infectionType.contains("edu") && ! infectionType.contains("work") && ! ( leavingPersonsActivity.contains("edu") && otherPersonsActivity.contains("edu") )){
+                    //edu can only interact with work or edu
+                    continue;
+                }
+
+                trackContactPerson(personLeavingContainer, otherPerson, leavingPersonsActivity);
```

Comment:
You changed the order of this block, which is now before `isRelevantForInfecttionDynamics`. Thus, the results are different and the tests failing.

Reply:
The question is: which persons should be tracked? Probably that should be all persons that somebody is in contact with, i.e. contact persons that fullfill the conditions on activity types and are not in quarantine or hospital. The check whether the interaction is relevant for infection dynamic can be processed later.

I implemented the additional check, whether a person is traceable in the following commit.

In my opinion, the order is correct now. But please review and comment :)